buttons/src/main/java/com/decathlon/vitamin/buttons/VitaminIconButtons.kt

```diff
@@ -0,0 +1,185 @@
+package com.decathlon.vitamin.buttons
+
+import android.content.Context
+import android.util.AttributeSet
+import android.util.TypedValue
+import androidx.annotation.AttrRes
+import com.google.android.material.button.MaterialButton
+
+private fun Context.getDimensionPixelSizeFromAttr(@AttrRes attributeId: Int): Int {
+    val typedValue = TypedValue().also {
+        theme.resolveAttribute(attributeId, it, true)
+    }
+    return resources.getDimensionPixelSize(typedValue.resourceId)
+}
+
+private val ICON_BUTTON_MEDIUM_ATTR = R.attr.vtmnIconButtonMediumSize
+private val ICON_BUTTON_LARGE_ATTR = R.attr.vtmnIconButtonLargeSize
+
+open class VitaminPrimaryMediumIconButton @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = R.attr.vtmnIconButtonPrimaryMediumStyle
+) : MaterialButton(context, attrs, defStyleAttr) {
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+        layoutParams.width = context.getDimensionPixelSizeFromAttr(ICON_BUTTON_MEDIUM_ATTR)
```

Comment:
Why is this needed? The XML declaration is not working as usual?

Reply:
No. If you don't force the size here, the button add padding of the text around the icon.

![Capture d’écran 2022-06-23 à 15 07 25](https://user-images.githubusercontent.com/1913901/175305871-57ce88d2-809a-4d99-8ba8-5c4ae2374711.png)

Reply:
So these lines in the ReadMe are wrong, isn't it? 

```xml
<com.google.android.material.button.MaterialButton
    style="?attr/vtmnIconButtonPrimaryMediumStyle"
    android:layout_width="?attr/vtmnIconButtonMediumSize"
    android:layout_height="?attr/vtmnIconButtonMediumSize"
    app:icon="@drawable/ic_vtmn_add" />
```

It will render the button the same as in your screenshot above?

Reply:
Not exactly because I give the `layout_width` and the `layout_height` too.